[RDY] vim-patch:8.0.{1006,1023,1029,1031,1040}

janlazo opened the pull request:
**vim-patch:8.0.1006: quickfix list changes when parsing text with 'erroformat'**

Problem:    Cannot parse text with 'erroformat' without changing a quickfix
            list.
Solution:   Add the "text" argument to getqflist(). (Yegappan Lakshmanan)
https://github.com/vim/vim/commit/7adf06f4e25c795ba32ff0b2e8591330f6a41afb

**vim-patch:8.0.1023: it is not easy to identify a quickfix list**

Problem:    It is not easy to identify a quickfix list.
Solution:   Add the "id" field. (Yegappan Lakshmanan)
vim/vim@a539f4f

**vim-patch:8.0.1029: return value of getqflist() is inconsistent**

Problem:    Return value of getqflist() is inconsistent.  (Lcd47)
Solution:   Always return an "items" entry.
vim/vim@da73253

**vim-patch:8.0.1031: "text" argument for getqflist() is confusing**

Problem:    "text" argument for getqflist() is confusing. (Lcd47)
Solution:   Use "lines" instead. (Yegappan Lakshmanan)
vim/vim@2c809b7

**vim-patch:8.0.1040: cannot use another error format in getqflist()**

Problem:    Cannot use another error format in getqflist().
Solution:   Add the "efm" argument to getqflist(). (Yegappan Lakshmanan)
vim/vim@3653822

janlazo committed:
vim-patch:8.0.1006: quickfix list changes when parsing text with 'erroformat'

Problem:    Cannot parse text with 'erroformat' without changing a quickfix
            list.
Solution:   Add the "text" argument to getqflist(). (Yegappan Lakshmanan)
https://github.com/vim/vim/commit/7adf06f4e25c795ba32ff0b2e8591330f6a41afb

janlazo commented on src/nvim/quickfix.c:
```diff
@@ -4109,6 +4113,34 @@ enum {
+      || (di->di_tv.v_type == VAR_LIST
+          && di->di_tv.vval.v_list != NULL)) {
+    qf_info_T *qi = xmalloc(sizeof(*qi));
+    memset(qi, 0, sizeof(*qi));
```
Why doesn't Vim use `calloc` here? Is it buggy or unavailable on some platforms?

justinmk replied:
@janlazo we use `xcalloc` in some cases, but it probably doesn't matter since the compiler optimizes it anyway.

janlazo commented on src/nvim/quickfix.c:
```diff
@@ -4031,18 +4031,22 @@ static void unload_dummy_buffer(buf_T *buf, char_u *dirname_start)
 /// Add each quickfix error to list "list" as a dictionary.
 /// If qf_idx is -1, use the current list. Otherwise, use the specified list.
-int get_errorlist(win_T *wp, int qf_idx, list_T *list)
+int get_errorlist(const qf_info_T *qi_arg, win_T *wp, int qf_idx, list_T *list)
```
Yes.

janlazo committed:
vim-patch:8.0.1023: it is not easy to identify a quickfix list

Problem:    It is not easy to identify a quickfix list.
Solution:   Add the "id" field. (Yegappan Lakshmanan)
https://github.com/vim/vim/commit/a539f4f1ae4a2b3a7dfce89cd3800214c9e990cf

janlazo committed:
vim-patch:8.0.1029: return value of getqflist() is inconsistent

Problem:    Return value of getqflist() is inconsistent.  (Lcd47)
Solution:   Always return an "items" entry.
https://github.com/vim/vim/commit/da73253a0b908bad03ddcd625fe3fb32008efbf6

janlazo committed:
vim-patch:8.0.1031: "text" argument for getqflist() is confusing

Problem:    "text" argument for getqflist() is confusing. (Lcd47)
Solution:   Use "lines" instead. (Yegappan Lakshmanan)
https://github.com/vim/vim/commit/2c809b7c7d2bb5e4b7fd09c3d312cadecf0c1ff0

janlazo committed:
lint

janlazo committed:
vim-patch:8.0.1040: cannot use another error format in getqflist()

Problem:    Cannot use another error format in getqflist().
Solution:   Add the "efm" argument to getqflist(). (Yegappan Lakshmanan)
https://github.com/vim/vim/commit/3653822546fb0f1005c32bb5b70dc9bfacdfc954

janlazo committed:
oldtests: win: fix buffer pathsep
